pkg/idp/third_party.go

```diff
@@ -55,14 +55,14 @@ type Configuration struct {
 	// AuthURL is the authorize url, typically something like: https://example.org/oauth2/auth
 	// Should only be used when the OAuth2 / OpenID Connect server is not supporting OpenID Connect Discovery and when
 	// `provider` is set to `generic`.
-	// validate that this field is required only when Provider field == "generic"
-	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_if=Provider generic"`
+	// validate that this field is required only when Provider field == "generic" and IssuerURL is empty
+	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_unless=IssuerURL,required_if=Provider generic"`
 
 	// TokenURL is the token url, typically something like: https://example.org/oauth2/token
 	// Should only be used when the OAuth2 / OpenID Connect server is not supporting OpenID Connect Discovery and when
 	// `provider` is set to `generic`.
-	// validate that this field is required only when Provider field == "generic"
-	TokenURL string `json:"token_url" yaml:"token_url" validate:"required_if=Provider generic"`
+	// validate that this field is required only when Provider field == "generic" and IssuerURL is empty
+	TokenURL string `json:"token_url" yaml:"token_url" validate:"required_unless=IssuerURL,required_if=Provider generic"`
```

Comment:
I believe [`required_without`](https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Required_Without) is the validation rule that should be used instead of `required_unless`. The `required_unless` rule [needs](https://github.com/canonical/identity-platform-admin-ui/actions/runs/11801933476/job/32876428929?pr=456#step:6:40) an exact value of the other field.

Unfortunately, there might still be a logical issue here. Let's take a look at the following example:
```
IssuerURL: some.issuer/url
AuthURL: nil
TokenURL: nil
Provider: generic
```

Having the TokenURL validation rule `required_without=IssuerURL,required_if=Provider generic`:
* `required_without=IssuerURL` check will pass. TokenURL isn't required, as IssuerURL is set.
* `required_if=Provider generic` check will fail. TokenURL is always required when Provider set to `generic`.

My domain knowledge is limited. But from the first glance, the flexibility of validation rules might not be enough for our case. Custom validation implementation might be an answer.

Reply:
FYI @nsklikas

Reply:
I have not tried it, but I don't understand it. In this case:

 ```
    IssuerURL: some.issuer/url 
    AuthURL: nil
    TokenURL: nil
    Provider: generic
```
> Having the TokenURL validation rule required_without=IssuerURL,required_if=Provider generic:
    - required_without=IssuerURL check will pass. TokenURL isn't required, as IssuerURL is set.
    - required_if=Provider generic check will fail. TokenURL is always required when Provider set to generic.

`required_without=IssuerURL` should fail, so the second rule shouldn't be considered. This means that it should be evaluated as required. From reading the docs I would expect that chaining conditions would happen as an `AND` not an `OR`. Am I missing something?

Reply:
If it turns out too complicated, let's park it and move it to a different PR.

Reply:
It works as an AND. I'm experimenting with a or also (`|`) but still something is off.
I'll check the custom validation func way